### PR TITLE
Battery handling

### PR DESCRIPTION
**Pull request by infirit**
Eeuh, sorry, I got a bit carried away experimenting and accidentally did the decoupling for you :sweat_smile: .

### CODE REVIEW

**Comment by infirit** [comment]
Also fixed the missing signal bars in [b54e707](https://github.com/blueman-project/blueman/pull/1734/commits/b54e70720f1fb4409c46bccdcf8cfa285c1975e1)

File: blueman/gui/manager/ManagerDeviceList.py [hunk 1]
```diff
@@ -450,15 +466,17 @@ def _update_power_levels(self, tree_iter: Gtk.TreeIter, device: Device, cinfo: c
...
+        if obj_path in self._batteries:
+            bars["battery"] = self._batteries[obj_path]["Percentage"]
 
         # cinfo init may fail for bluetooth devices version 4 and up
         # FIXME Workaround is horrible and we should show something better
         if cinfo.failed:
             if not bars:
                 bars = {"rssi": 100.0, "tpl": 100.0, "lq": 100.0}
+            else:
+                bars.update({"rssi": 100.0, "tpl": 100.0, "lq": 100.0})
```
Comment (cschramm) [inline blueman/gui/manager/ManagerDeviceList.py:479]: Could be just the `else` branch, no?

File: blueman/plugins/applet/ConnectionNotifier.py [hunk 2]
```diff
@@ -13,10 +16,23 @@ class ConnectionNotifier(AppletPlugin):
     __description__ = _("Shows desktop notifications when devices get connected or disconnected.")
 
     def on_device_property_changed(self, path: str, key: str, value: Any) -> None:
-        if key == "Connected":
-            device = Device(obj_path=path)
-            if value and "Battery" in self.parent.Plugins.get_loaded() and Battery.applicable(device):
-                return
+        def show_notification() -> bool:
+            try:
+                perc = f"{battery['Percentage']}%"
```
Comment (cschramm) [inline blueman/plugins/applet/ConnectionNotifier.py:21]: Merging the notifications deserves a changelog entry.

File: blueman/plugins/applet/ConnectionNotifier.py [hunk 3]
```diff
@@ -13,10 +16,23 @@ class ConnectionNotifier(AppletPlugin):
...
-                         icon_name=device["Icon"]).show()
+            txt = f"{_('Connected')} {perc if perc is not None else ''}"
+            Notification(device["Alias"], txt, icon_name=icon_name).show()
+            return False
+
+        device = Device(obj_path=path)
+        battery = Battery(obj_path=path)
+
+        if key == "Connected" and value:
+            GLib.timeout_add_seconds(5, show_notification)
```
Comment (cschramm) [inline blueman/plugins/applet/ConnectionNotifier.py:36]: I really don't like the generic delay but let's get this merged an I'll test a little bit with notification replacements on top of it (and for other topics).

**Review by cschramm — APPROVED** [review]

File: blueman/gui/manager/ManagerDeviceList.py (hunk 1 above)
Comment (infirit) [reply blueman/gui/manager/ManagerDeviceList.py:479]: we don't even need to branch, we can use dict.update in both cases. I'll fix that.

File: blueman/plugins/applet/ConnectionNotifier.py (hunk 2 above)
Comment (infirit) [reply blueman/plugins/applet/ConnectionNotifier.py:21]: Yes, will do.

File: blueman/gui/manager/ManagerDeviceList.py (hunk 1 above)
Comment (cschramm) [reply blueman/gui/manager/ManagerDeviceList.py:479]: That's what I tried to express. :sweat_smile:

File: blueman/plugins/applet/ConnectionNotifier.py (hunk 3 above)
Comment (cschramm) [reply blueman/plugins/applet/ConnectionNotifier.py:36]: The notification stuff itself is easy. `_NotificationBubble` actually already handles replacing. See #1736. On top of that we just need to keep track of the notification (with a device -> notification mapping I guess).

**Comment by infirit** [comment]
KDE was rightly to complain as I was sending disconnect signals for every property change. Fixed the last bits so this should be good now.

**Comment by infirit** [comment]
I'll cherry-pick the bug fixes for stable

**Review by cschramm — APPROVED** [review]

**Pull request merged**